core_lib/tool/selecttool.cpp

```diff
@@ -5,7 +5,9 @@
 #include "toolmanager.h"
 #include "selecttool.h"
 
-QPointF gSelectionOrigin;
+// Store selection origin so we can calculate
+// the selection rectangle in mousePressEvent.
+static QPointF gSelectionOrigin;
```

Comment:
Agree with @CandyFace, this is not necessary. However I'll merge the code anyway and refactor later.